packages/manager/apps/dedicated/client/app/ip/byoip/byoip.service.js

```diff
@@ -0,0 +1,58 @@
+export default class ByoipService {
+  /* @ngInject */
+  constructor($http, $q, coreConfig, User) {
+    this.$http = $http;
+    this.$q = $q;
+    this.User = User;
+    this.coreConfig = coreConfig;
+  }
+
+  getCatalog() {
+    this.user = this.coreConfig.getUser();
+    return this.$http
+      .get('/order/catalog/formatted/bringYourOwnIp', {
+        params: {
+          ovhSubsidiary: this.user.ovhSubsidiary,
+        },
+      })
+      .then(({ data }) => {
+        return data.plans[0];
+      });
+  }
+
+  getToken(region) {
+    return this.$http
+      .get('/me/bringYourOwnIp/token', {
+        params: {
+          campus: region,
+        },
+        cache: this.cache,
+      })
+      .then(({ data }) => data);
+  }
+
+  /**
+   * Redirect to the express order page
+   * @param {Object} plan [detials of the plan]
+   * @param {config} array [configuration of the plan like name, region and others]
+   */
+  gotToExpressOrder(plan, config) {
+    const params = [
+      {
+        planCode: plan.planCode,
+        configuration: config,
+        option: [],
+        quantity: 1,
+        productId: 'bringYourOwnIp',
+      },
+    ];
+
+    return this.goToExpressOrderUrl(params);
+  }
+
+  goToExpressOrderUrl(payload) {
```

Comment:
looks like you are not redirecting to express order. in that case rename the function to getExpressOrder